editor/writing-flow/index.js

```diff
@@ -37,31 +45,55 @@ class WritingFlow extends Component {
 			) );
 	}
 
-	moveFocusInContainer( target, direction = 'UP' ) {
-		const focusableNodes = this.getVisibleTabbables();
-		if ( direction === 'UP' ) {
-			focusableNodes.reverse();
+	getClosestTabbable( target, isReverse ) {
+		let focusableNodes = this.getVisibleTabbables();
+
+		if ( isReverse ) {
+			focusableNodes = reverse( focusableNodes );
 		}
 
-		const targetNode = focusableNodes
-			.slice( focusableNodes.indexOf( target ) )
-			.reduce( ( result, node ) => {
-				return result || ( node.contains( target ) ? null : node );
-			}, null );
+		focusableNodes = focusableNodes.slice( focusableNodes.indexOf( target ) );
 
-		if ( targetNode ) {
-			placeCaretAtEdge( targetNode, direction === 'DOWN' );
-		}
+		return find( focusableNodes, ( node, i, array ) => {
+			if ( node.contains( target ) ) {
+				return false;
+			}
+
+			const nextNode = array[ i + 1 ];
+
+			// Skip node if it contains a focusable node.
+			if ( nextNode && node.contains( nextNode ) ) {
+				return false;
+			}
+
+			return true;
+		} );
 	}
 
 	onKeyDown( event ) {
 		const { keyCode, target } = event;
```

Comment:
I missed this until now, but FYI [`KeyboardEvent.keyCode` has been deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode) and `KeyboardEvent.key` is to be preferred.

Something to revise in the future?

Reply:
Probably. We've been using it in Gutenberg quite a lot, so does TinyMCE. Maybe better to discuss separately.

Reply:
Previously I'd avoided `key` because of browser support; I don't recall if this is specifically an older IE thing, or due to the non-standard behavior described in the "Notes" at:

http://caniuse.com/#feat=keyboardevent-key